base/logging.jl

```diff
@@ -278,19 +278,19 @@ default_group(file) = Symbol(splitext(basename(file))[1])
 function logmsg_code(_module, file, line, level, message, exs...)
     log_data = process_logmsg_exs(_module, file, line, level, message, exs...)
     quote
-        level = $level
-        std_level = convert(LogLevel, level)
+        local level = $level
+        local std_level = convert(LogLevel, level)
         if std_level >= getindex(_min_enabled_level)
-            group = $(log_data._group)
-            _module = $(log_data._module)
-            logger = current_logger_for_env(std_level, group, _module)
+            local group = $(log_data._group)
+            local _module = $(log_data._module)
+            local logger = current_logger_for_env(std_level, group, _module)
             if !(logger === nothing)
-                id = $(log_data._id)
+                local id = $(log_data._id)
                 # Second chance at an early bail-out (before computing the message),
                 # based on arbitrary logger-specific logic.
                 if _invoked_shouldlog(logger, level, _module, group, id)
-                    file = $(log_data._file)
-                    line = $(log_data._line)
+                    local file = $(log_data._file)
+                    local line = $(log_data._line)
```

Comment:
guess we can wrap the entire block into `let` instead of adding bunch of `local`s ? I hope it would look simpler